src/core/tests/preprocess.cpp

```diff
@@ -914,6 +914,22 @@ TEST(pre_post_process, mean_vector_dynamic_channels_shape) {
     EXPECT_EQ(f->get_output_element_type(0), element::f32);
 }
 
+TEST(pre_post_process, pad_vector_constant_layout) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 199, 199});
+    auto p = PrePostProcessor(f);
+    
+    std::stringstream exp_dump;
+    exp_dump << PartialShape{1, 3, 200, 200};
+    try{
+        p.input().preprocess().pad({0, 0, 0, 0}, {0, 0, 1, 1}, 0);
+        p.build();
```

Comment:
can we also compare shapes and check that padding is applied? 
or even better - compare with reference function

Reply:
I think in this test shapes have been checked(find expected shape in error because it won't match next layer)
Besides I added out_of_range and dim_mismatch check, do I require to incorporate any additional examinations?